### PR TITLE
Add `listServer` API to director Web UI

### PR DESCRIPTION
**Pull request by haoming29**
As part of the implementation to #265, this PR introduced a new API for director Web UI to get a list of advertised origins and caches: 

`/api/v1.0/director_ui/servers?server_type=origin` or `=cache` or without query parameter to get all servers advertised to the director.

Response:

```json
[
  {
    "name": "fae8c2865de4",
    "authUrl": "https://fae8c2865de4:8443",
    "url": "https://fae8c2865de4:8443",
    "webUrl": "https://fae8c2865de4:8444",
    "type": "Origin",
    "latitude": 0,
    "longitude": 0
  }
]
```

### CODE REVIEW

**Comment by haoming29** [comment]
@CannonLock here's the API to for querying and get a list of origins and caches registered at the director (which is different from what's showing up in the namespace registry UI)

**Comment by haoming29** [comment]
> @haoming29 Should this be added to the openapi spec?

I'm holding off adding new APIs to the spec as @bbockelm mentioned that we want to introduce the automated spec generation with annotations. I'm not sure what's the timeline for that though.

File: director/director_ui.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import (
+	"net/http"
+	"strings"
+
+	"github.com/gin-gonic/gin"
+)
+
+type (
+	listServerRequest struct {
+		ServerType string `form:"server_type,omitempty"` // "cache" or "origin"
```
Comment (turetske) [inline director/director_ui.go:12]: What is this omit empty doing?

**Review by turetske — CHANGES_REQUESTED** [review]

File: director/director_ui.go (hunk 1 above)
Comment (haoming29) [reply director/director_ui.go:12]: Thanks for pointing out. It should be removed it as this is a request struct not response

**Comment by CannonLock** [comment]
@haoming29 Are the lat lon 0 because I don't have the Maxmind db?

**Comment by haoming29** [comment]
> @haoming29 Are the lat lon 0 because I don't have the Maxmind db?

Unfortunately for Pelican Origins, we don't specify Lat/Long when we advertise to the director

**Comment by CannonLock** [comment]
> > @haoming29 Are the lat lon 0 because I don't have the Maxmind db?
> 
> Unfortunately for Pelican Origins, we don't specify Lat/Long when we advertise to the director. I believe this field is only populated when we fetch origins from OSDF for now

Is it a new ticket to have unset values come in as null? null is a better proxy for a unset then 0.

**Review by turetske — APPROVED** [review]
LGTM!

**Pull request merged**